feature/security/gnsi/credentialz/tests/ssh_public_key_authentication/ssh_public_key_authentication_test.go

```diff
@@ -0,0 +1,280 @@
+package ssh_public_key_authentication
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	username                    = "testuser"
+	userPrivateKeyFilename      = "user"
+	userPublicKeyFilename       = "user.pub"
+	authorizedKeysListVersion   = "v1.0"
+	authorizedKeysListCreatedOn = 1705962293
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func prepareUserKey(t *testing.T, dir string) {
```

Comment:
same comment as (I think) credentialz-3 "this all should go into a utility library that all of this code can use"

Reply:
Added common library in `internal/security/credz/credz.go`